packages/phalanx/example/FiniteElementAssembly_NoTags/FiniteElementAssembly.cpp

```diff
@@ -0,0 +1,311 @@
+      fm.registerEvaluator<Jacobian>(j);
+    }
+
+    // Project DOF Gradients to qps
+    for (int eq=0; eq < num_equations; ++eq) {
+      std::stringstream s;
+      s << "equation_" << eq;
+      RCP<ProjectGradientToQP<Residual,MyTraits>> r =
+        rcp(new ProjectGradientToQP<Residual,MyTraits>(s.str(),basis_layout,grad_qp_layout));
+      fm.registerEvaluator<Residual>(r);
+      RCP<ProjectGradientToQP<Jacobian,MyTraits>> j =
+        rcp(new ProjectGradientToQP<Jacobian,MyTraits>(s.str(),basis_layout,grad_qp_layout));
+      fm.registerEvaluator<Jacobian>(j);
+    }
+
+    // source term
+    {
+      RCP<Constant<Residual,MyTraits>> r = rcp(new Constant<Residual,MyTraits>("s_dot",qp_layout,5.0));
+      fm.registerEvaluator<Residual>(r);
+      RCP<Constant<Jacobian,MyTraits>> j = rcp(new Constant<Jacobian,MyTraits>("s_dot",qp_layout,5.0));
+      fm.registerEvaluator<Jacobian>(j);
+    }
+    
+    // Zero contributed field
+    for (int eq=0; eq < num_equations; ++eq) {
+      std::stringstream s;
+      s << "residual_" << eq;
+
+      RCP<ZeroContributedField<Residual,MyTraits>> r =
+        rcp(new ZeroContributedField<Residual,MyTraits>(s.str(),basis_layout));
+      fm.registerEvaluator<Residual>(r);
+
+      RCP<ZeroContributedField<Jacobian,MyTraits>> j =
+        rcp(new ZeroContributedField<Jacobian,MyTraits>(s.str(),basis_layout));
+
+      fm.registerEvaluator<Jacobian>(j);
+    }
+    
+    // Integrate source term
+    for (int eq=0; eq < num_equations; ++eq) {
+      std::stringstream s;
+      s << "s_dot";
+      std::stringstream s_r;
+      s_r << "residual_" << eq;
+      
+      RCP<IntegrateSourceTerm<Residual,MyTraits>> r =
+        rcp(new IntegrateSourceTerm<Residual,MyTraits>(s.str(),qp_layout,s_r.str(),basis_layout));
+
+      fm.registerEvaluator<Residual>(r);
+      
+      RCP<IntegrateSourceTerm<Jacobian,MyTraits>> j =
+        rcp(new IntegrateSourceTerm<Jacobian,MyTraits>(s.str(),qp_layout,s_r.str(),basis_layout));
+
+      fm.registerEvaluator<Jacobian>(j);
+    }
+
+    // Integrate diffusion term
+    for (int eq=0; eq < num_equations; ++eq) {
+      std::stringstream s;
+      s << "equation_" << eq;
+      std::stringstream s_r;
+      s_r << "residual_" << eq;
+      
+      RCP<IntegrateDiffusionTerm<Residual,MyTraits>> r =
+        rcp(new IntegrateDiffusionTerm<Residual,MyTraits>(s.str(),grad_qp_layout,s_r.str(),basis_layout));
+
+      fm.registerEvaluator<Residual>(r);
+      
+      RCP<IntegrateDiffusionTerm<Jacobian,MyTraits>> j =
+        rcp(new IntegrateDiffusionTerm<Jacobian,MyTraits>(s.str(),grad_qp_layout,s_r.str(),basis_layout));
+
+      fm.registerEvaluator<Jacobian>(j);
+    }
+
+    // Scatter DOFs
+    for (int eq=0; eq < num_equations; ++eq) {
+      std::stringstream s;
+      s << "residual_" << eq;
+      
+      RCP<PHX::FieldTag> scatter_tag_r = rcp(new Tag<MyTraits::Residual::ScalarT>(s.str(),scatter_layout));
```

Comment:
From a testing as documentation perspective it might be nice to move lines 117 and 118 closer to here so that people can see the association with the residual and jacobian declaration (I looked for this for a few minutes).